### PR TITLE
solveset(Abs(x) + 1, x) now returns EmptySet() typo fixed

### PR DESCRIPTION
**Pull request by gxyd**
Earlier

```
>>> x = Symbol('x', real=True)
>>> b = Symbol('b', real=True, positive=True)
>>> solveset(Abs(x) + b, x)
{b, -b}
>>> solveset(Eq(Abs(x), -1), x)
False
```

Now

```
>>> x = Symbol('x', real=True)
>>> b = Symbol('b', real=True, positive=True)
>>> solveset(Abs(x) + b, x)
EmptySet()
>>> solveset(Eq(Abs(x), -1), x)
EmptySet()
```

Fixes issue #9556 and typo fixed.

### CODE REVIEW

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -854,6 +862,9 @@ def solveset(f, symbol=None):
 
     f = sympify(f)
 
+    if f is S.false:
```
Comment (debugger22) [inline sympy/solvers/solveset.py]: Perhaps, just `if not f`. Why did you choose `BooleanFalse`?

File: sympy/solvers/tests/test_solveset.py [hunk 2]
```diff
@@ -814,3 +814,10 @@ def test_issue_9522():
 
     assert solveset(expr1, x) == EmptySet()
     assert solveset(expr2, x) == EmptySet()
+
```
Comment (debugger22) [inline sympy/solvers/tests/test_solveset.py]: One more line.

File: sympy/solvers/solveset.py (hunk 1 above)
Comment (gxyd) [inline sympy/solvers/solveset.py]: Reason is `solve(0, x)`  returns `[ ]` but `solveset(0, x)` returns `(-oo, oo)` . At  the top of `solveset` module it is written that `solveset` returns a `set` not mentioning of boolean.

Comment (debugger22) [inline sympy/solvers/solveset.py]: I was talking about the boolean in the conditional.

Comment (gxyd) [inline sympy/solvers/solveset.py]: That is what i said.
if i use `if not f`
```
>>> solveset(0, x)
EmptySet()               # this is returned
```

if i use `if f is S.false`
```
>>> solveset(0, x)
(-oo, oo)
```
Hope this clarifies ?

File: sympy/solvers/tests/test_solveset.py (hunk 2 above)
Comment (gxyd) [inline sympy/solvers/tests/test_solveset.py]: Done.

**Comment by gxyd** [comment]
ping @hargup , as you said to me, for this PR.

**Comment by aktech** [comment]
@gxyd  Please fix Merge Conflicts as well.

File: sympy/solvers/solveset.py [hunk 3]
```diff
@@ -76,6 +76,14 @@ def _invert_real(f, g_ys, symbol):
                             imageset(Lambda(n, f.inverse()(n)), g_ys), symbol)
 
     if isinstance(f, Abs):
+        _a = []
+        for i in g_ys:
+            try:
+                if i < 0:
```
Comment (gxyd) [inline sympy/solvers/solveset.py]: After discussion on #9586 i think,  we should do it the way it is done in second commit. Since for after my second commit, this is the thing
```
>>> x = Symbol('x', real=True)
>>> y = Symbol('y')
>>> invert_real(Abs(x) + 1, y, x)
(x, {-y + 1, y - 1})
```

**Comment by gxyd** [comment]
@aktech  @hargup  please review.

File: sympy/solvers/solveset.py [hunk 4]
```diff
@@ -855,6 +863,9 @@ def solveset(f, symbol=None):
 
     f = sympify(f)
 
+    if f is S.false:
+        return EmptySet()
```
Comment (aktech) [inline sympy/solvers/solveset.py]: The above two lines are redundant.

File: sympy/solvers/tests/test_solveset.py [hunk 5]
```diff
@@ -873,3 +873,12 @@ def test_linsolve():
...
                     (f - c*e/a)/(d - b*c/a)))
     assert linsolve(system, [x, y]) == sol
+
+
+def test_issue_9556():
+    x = Symbol('x', real=True)
+    b = Symbol('b', real=True, positive=True)
+
+    assert solveset(Abs(x) + 1, x) == EmptySet()
+    assert solveset(Eq(Abs(x), -1), x) == EmptySet()
```
Comment (aktech) [inline sympy/solvers/tests/test_solveset.py]: This is a repetition of the above L882.

File: sympy/solvers/solveset.py [hunk 6]
```diff
@@ -76,6 +76,14 @@ def _invert_real(f, g_ys, symbol):
...
 
     if isinstance(f, Abs):
+        _a = []
+        for i in g_ys:
+            try:
+                if i < 0:
+                    _a.append(i)
+            except TypeError:
+                pass
+        g_ys = g_ys - FiniteSet(*_a)
```
Comment (aktech) [inline sympy/solvers/solveset.py]: Perhaps, you can simply do this:

```diff
diff --git a/sympy/solvers/solveset.py b/sympy/solvers/solveset.py
index 4c1507f..04e9d16 100644
--- a/sympy/solvers/solveset.py
+++ b/sympy/solvers/solveset.py
@@ -76,14 +76,7 @@ def _invert_real(f, g_ys, symbol):
                             imageset(Lambda(n, f.inverse()(n)), g_ys), symbol)
 
     if isinstance(f, Abs):
-        _a = []
-        for i in g_ys:
-            try:
-                if i < 0:
-                    _a.append(i)
-            except TypeError:
-                pass
-        g_ys = g_ys - FiniteSet(*_a)
+        g_ys = g_ys - FiniteSet(*[i for i in g_ys if i.is_negative])

```

File: sympy/solvers/solveset.py (hunk 4 above)
Comment (gxyd) [inline sympy/solvers/solveset.py]: Currently:
```
>>> x = Symbol('x', positive=True)
>>> solveset(Eq(x, -1), x)
False
```
But Is not `solveset` supposed to return always a `set` ? That's why i did this.

Comment (aktech) [inline sympy/solvers/solveset.py]: > But Is not solveset supposed to return always a set ?

Yes, perhaps this is a good catch, but not related to this PR. This could be incorporated in a different commit.

File: sympy/solvers/solveset.py (hunk 6 above)
Comment (gxyd) [inline sympy/solvers/solveset.py]: Probably i regret about making such a silly mistake, i could surely make this better.

**Comment by gxyd** [comment]
@aktech now Please have a look.

File: sympy/solvers/tests/test_solveset.py [hunk 7]
```diff
@@ -873,3 +873,12 @@ def test_linsolve():
     sol = FiniteSet((-b*(f - c*e/a)/(a*(d - b*c/a)) + e/a,
                     (f - c*e/a)/(d - b*c/a)))
     assert linsolve(system, [x, y]) == sol
+
+
+def test_issue_9556():
+    x = Symbol('x', real=True)
+    b = Symbol('b', real=True, positive=True)
```
Comment (aktech) [inline sympy/solvers/tests/test_solveset.py]: The one which is `positive` is `real` as well.

**Comment by aktech** [comment]
It looks good to me, except for the [minor comment](https://github.com/sympy/sympy/pull/9570#discussion_r33756141) , the code may change in future while fixing #9588 , but since the scope of this PR is to fix #9556 which it does, I am **+1** to this.

File: sympy/solvers/tests/test_solveset.py (hunk 7 above)
Comment (aktech) [inline sympy/solvers/tests/test_solveset.py]:
```python
In [10]: b = Symbol('b', positive=True)

In [11]: b.is_real
Out[11]: True
```
So, `b = Symbol('b', positive=True)` would work as well.

Comment (gxyd) [inline sympy/solvers/tests/test_solveset.py]: Done.

**Comment by aktech** [comment]
Will commit in 24hr if there is no objection from @hargup

**Comment by hargup** [comment]
+1

**Comment by gxyd** [comment]
Thanks @hargup  and escpecially @aktech :)
